caffeine@patapon.info/extension.js

```diff
@@ -260,70 +648,124 @@ class Caffeine extends PanelMenu.Button {
     }
 
     _inhibitorRemoved(proxy, sender, [object]) {
-        let index = this._objects.indexOf(object);
-        if (index !== -1) {
-            if (this._apps[index] === 'user')
-                this._saveUserState(false);
-            // Remove app from list
-            this._apps.splice(index, 1);
-            this._cookies.splice(index, 1);
-            this._objects.splice(index, 1);
-            if (this._apps.length === 0) {
-                this._state = false;
-                this._icon.gicon = Gio.icon_new_for_string(`${Me.path}/icons/${DisabledIcon}.svg`);
-                if (this._settings.get_boolean(SHOW_NOTIFICATIONS_KEY))
-                    this._sendNotification('disabled');
+        // Get the first removed request 
+        let requested_id = this._inhibition_removed_fifo.shift();
+
+        if(requested_id){
+            let appData = this._appInhibitedData.get(requested_id);        
+            if (appData){
+                if (requested_id === 'user')
+                       this._saveUserState(false);
```

Comment:
NIT: indent 
```suggestion
                if (requested_id === 'user') {
                    this._saveUserState(false);
                }
```

Reply:
I changed it. 

```
                if (requested_id === 'user')
                       this._saveUserState(false);
```
Is this syntax better to avoid ? 
I thought it was just a more compact way to use when having only one short line of code.

Reply:
It's fine. Saves some lines but sometimes it think it's harder to read